docs/source/advanced/contrib_thunder.rst

```diff
@@ -0,0 +1,468 @@
+example of ``clang.slice_in_dim``
+
+.. code-block:: python
+  :lineno-start: 938
+
+  @clangop()
+  def slice_in_dim(a, start_index, limit_index, stride=1, dim=0):
+      len_dim = a.shape[dim]
+      start_index = utils.canonicalize_dim_idx(len_dim, start_index)
+      limit_index = utils.canonicalize_dim_idx(len_dim, limit_index)
+
+      # Handles the start idx being greater than the dimension length by returning
+      #   a tensor with no elements
+      if start_index >= len_dim:
+          shape = list(a.shape)
+          shape[dim] = 0
+          return full(shape, 0, device=a.device, dtype=a.dtype)
+
+      # Handles the limit idx being greater than the dimension length by clamping it
+      if limit_index >= len_dim:
+          limit_index = len_dim
+
+      # Constructs args for the slice prim
+      start_indices = [0] * a.ndim
+      limit_indices = list(a.shape)
+      strides = [1] * a.ndim
+
+      start_indices[dim] = start_index
+      limit_indices[dim] = limit_index
+      strides[dim] = stride
+
+      return prims.slice_prim(a, start_indices, limit_indices, strides)
+
+enhances ``prims.slice_prim`` by adding support for default arguments and negative dimensions, among others.
+
+Once the symbol is created, we need to tell *Thunder* how to *execute* it.
+Since we are updating the ``torch`` executor, the following lines are added to the
+`executors/torchex.py <https://github.com/Lightning-AI/lightning-thunder/blob/main/thunder/executors/torchex.py>`_ file
+
+.. code-block:: python
+   :emphasize-lines: 2
+   :lineno-start: 465
+
+   unbind = _register_torch_operation("unbind")
+   unfold = _register_torch_operation("unfold", module=torch.Tensor)
+   unsqueeze = _register_torch_operation("unsqueeze")
+
+.. code-block:: python
+   :emphasize-lines: 2
+   :lineno-start: 536
+
+   _register_implementation(prims.transpose, checker=_always_executable, execution_transform=_transpose_prim_transform)
+   _register_implementation(prims.unfold, unfold, checker=_always_executable)
+   _register_implementation(prims.view, view, checker=_always_executable)
+
+the first one registers a new symbol that is directly tied to the ``torch.Tensor.unfold``, and the second
+associates this symbol with ``prims.unfold`` upon execution unless the ``checker`` fails
+(``checker`` is useful, for example, for checking PyTorch versions).
+Having ``checker=_always_executable`` always greenlights this association, and, hence,
+whenever the ``torch`` executor tries to execute ``prims.unfold``, it executes ``torch.Tensor.unfold``.
+
+The mapping of the ``prims.unfold`` symbol to ``torch.Tensor.unfold`` is very simple since the inputs
+to ``prims.unfold`` can directly be passed to ``torch.Tensor.unfold`` without any additional pre-preprocessing
+(association between ``TensorProxy`` and ``torch.Tensor`` is handled automatically by the ``torch`` executor).
+This is not the case with any operation, however, and sometimes the symbol's interface has to undergo
```

Comment:
Did you mean?

```suggestion
This is not the case with all operations, however, and sometimes the symbol's interface has to undergo
```

Reply:
Sorry, are these not equivalent?

Reply:
My non-english-native brain prefers "all", but feel free to keep yours